vmm/vmm.go

```diff
@@ -99,17 +99,23 @@ func (v *VMM) Setup() error {
 func (v *VMM) Boot() error {
 	var err error
 
-	var wg sync.WaitGroup
-
 	trace := v.TraceCount > 0
 	if err := v.SingleStep(trace); err != nil {
 		return fmt.Errorf("setting trace to %v:%w", trace, err)
 	}
 
+	g := new(errgroup.Group)
```

Comment:
It would be better to create the context here (e.g, `errgroup.WithContext()`), and in addition, for functions that run inside goroutine, such as Machine.VCPU() and Serial.Start(), the context should be taken as an argument. (so that it can be politely cancelled).